docs/reference/analysis/tokenizers/pathhierarchy-tokenizer-examples.asciidoc

```diff
@@ -0,0 +1,195 @@
+[[analysis-pathhierarchy-tokenizer-examples]]
+=== Example tokenizing file paths
+
+A common use-case for the `path_hierarchy` tokenizer is filtering results by 
+file paths. If indexing a file path along with the data, the use of the 
+`path_hierarchy` tokenizer to analyze the path allows filtering the results 
+by different parts of the file path string.
+
+
+This example configures an index to have two custom analyzers and applies
+those analyzers to multifields of the `file_path` text field that will 
+store filenames. One of the two analyzers uses reverse tokenization.
+Some sample documents are then indexed to represent some file paths
+for photos inside photo folders of two different users.
+
+
+[source,js]
+--------------------------------------------------
+PUT /file-path-test/
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "custom_path_tree": {
+          "tokenizer": "custom_hierarchy"
+        },
+        "custom_path_tree_reversed": {
+          "tokenizer": "custom_hierarchy_reversed"
+        }
+      },
+      "tokenizer": {
+        "custom_hierarchy": {
+          "type": "path_hierarchy",
+          "delimiter": "/",
+          "search_analyzer": "keyword"
```

Comment:
The first time I see `search_analyzer` parameter under `tokenizer`. Did you intend to put it under `mappings` for corresponding fields?

Reply:
I vaguely remember seeing something like this before where the nesting was wrong but wasn't caught and worked anyway, let me verify I'm doing this correctly and if we need a bug ticket for extra validation to prevent the "wrong" way perhaps.

Reply:
Yeah I think I've encountered this before perhaps, it appears to be ignored, you can put any text you want in there and it doesn't appear to validate:
```
  "settings": {
    "analysis": {
      "analyzer": {
        "custom_path_tree": {
          "tokenizer": "custom_hierarchy"
        },
        "custom_path_tree_reversed": {
          "tokenizer": "custom_hierarchy_reversed"
        }
      },
      "tokenizer": {
        "custom_hierarchy": {
          "type": "path_hierarchy",
          "delimiter": "/",
          "jimbob": "keyword"
        },
        "custom_hierarchy_reversed": {
          "type": "path_hierarchy",
          "delimiter": "/",
          "any_text_you_want": "keyword",
          "reverse": "true"
        }
      }
    }
  }
```

I'll log (or locate if I've already logged it) a bug ticket to address this.
For now, I'll take it out and nothing should functionally change and tests should work the same way.